### ISSUE TITLE
install fails when trying to install shoe/sockjs-client

### ISSUE THREAD

**Issue by ataner** [notwaldorf/tiny-care-terminal#131]
I can't install tiny-care-terminal using npm. I get the following error, although I have git-status installed and other npm packages install without issues. 

I have tried getting rid of node and reinstalling from scratch, but the error persists. I'm running MacOs High Sierra with the latest XCode and installed node via Homebrew

```npm install -g tiny-care-terminal
npm WARN deprecated phantom@0.8.4: v1 is no longer maintained, please upgrade to v2.0+ as soon possible.
npm WARN deprecated win-spawn@2.0.0: use [cross-spawn](https://github.com/IndigoUnited/node-cross-spawn) or [cross-spawn-async](https://github.com/IndigoUnited/node-cross-spawn-async) instead.
npm WARN deprecated node-uuid@1.3.3: Use uuid module instead
npm WARN tiny-care-terminal@1.3.2 requires a peer of git-standup@^2.1.8 but none is installed. You must install peer dependencies yourself.

npm ERR! path /Users/renata/.nvm/versions/node/v8.9.4/lib/node_modules/tiny-care-terminal/node_modules/shoe/node_modules/sockjs-client
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename
npm ERR! enoent ENOENT: no such file or directory, rename '/Users/renata/.nvm/versions/node/v8.9.4/lib/node_modules/tiny-care-terminal/node_modules/shoe/node_modules/sockjs-client' -> '/Users/renata/.nvm/versions/node/v8.9.4/lib/node_modules/tiny-care-terminal/node_modules/shoe/node_modules/.sockjs-client.DELETE'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent
```

[node.log](https://github.com/notwaldorf/tiny-care-terminal/files/1678827/node.log)

- **Node Version**: `v8.9.4`
- **tiny-care-terminal version**: ` ersion: '1.3.2',`
- **Shell**: `/bin/bash`
- **Terminal Program**: `iTerm2`
- **Operating System**: `MacOS High Sierra`

Thank you!

**Comment by kerseyd27** [comment]
+1

**Comment by builtbyproxy** [comment]
unrelated but install tiny-care-terminal BEFORE git status, fixed all module issues for me when i had them

worth a go

**Comment by mojoaxel** [comment]
@ataner please make sure you have a up-to-date version of node installed

**Comment by kerseyd27** [comment]
XXX:~ xxx$ npm install -g tiny-care-terminal
npm WARN deprecated phantom@0.8.4: v1 is no longer maintained, please upgrade to v2.0+ as soon possible.
npm WARN deprecated win-spawn@2.0.0: use [cross-spawn](https://github.com/IndigoUnited/node-cross-spawn) or [cross-spawn-async](https://github.com/IndigoUnited/node-cross-spawn-async) instead.
npm WARN deprecated node-uuid@1.3.3: Use uuid module instead
npm WARN tiny-care-terminal@1.3.2 requires a peer of git-standup@^2.1.8 but none is installed. You must install peer dependencies yourself.

npm ERR! path /usr/local/lib/node_modules/tiny-care-terminal/node_modules/shoe/node_modules/sockjs-client
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename
npm ERR! enoent ENOENT: no such file or directory, rename '/usr/local/lib/node_modules/tiny-care-terminal/node_modules/shoe/node_modules/sockjs-client' -> '/usr/local/lib/node_modules/tiny-care-terminal/node_modules/shoe/node_modules/.sockjs-client.DELETE'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/xxx/.npm/_logs/2018-02-02T20_14_51_019Z-debug.log
XXX:~ xxx$ node -v
v8.9.4

Same as OP, I have v8.9.4 of node installed

**Comment by builtbyproxy** [comment]
Have you tried my suggestion? I had the git standup warning and it fixed it

**Comment by kerseyd27** [comment]
> unrelated but install tiny-care-terminal BEFORE git status

> npm WARN tiny-care-terminal@1.3.2 requires a peer of git-standup@^2.1.8 but none is installed. You must install peer dependencies yourself.

Because the install of tiny-care-terminal failed, I never installed git-standup, and it currently isn't installed because I'm waiting until this installs

**Comment by kerseyd27** [comment]
What ended up working for me is deleting all files in node_modules that work with this project, downloading Xcode, and then running npm install -g tiny-care-terminal

What I found is that there was a partial download for git-standup even though I never ran an install for it, so I'm not sure if that step is necessary. I'm on OS X Sierra and was able to install then set the variables for configuration in bash_profile. I don't use git standup, I just use local repositories.

Hope this works for y'all too

**Comment by gmcdev** [comment]
Unrelated to tiny-care-terminal...having the same problem with sockjs-client on OS X Sierra:

npm ERR! path /Users/greg.connell/Documents/_RED_AD_TECH/wp-creative-server/node_modules/shoe/node_modules/sockjs-client
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename
npm ERR! enoent ENOENT: no such file or directory, rename '/Users/greg.connell/Documents/_RED_AD_TECH/wp-creative-server/node_modules/shoe/node_modules/sockjs-client' -> '/Users/greg.connell/Documents/_RED_AD_TECH/wp-creative-server/node_modules/shoe/node_modules/.sockjs-client.DELETE'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

**Comment by veekas** [comment]
I'm getting a slightly different error message, but with the same `sockjs-client` package.

**node**: v8.10.0
**npm**: 5.7.1

**error**:
```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for sockjs-client@0.0.0-unreleasable
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget
npm ERR! notarget It was specified as a dependency of 'shoe'
```

**relevant logs**:
```
...
523 silly resolveWithNewModule sockjs@0.3.7 checking installable status
524 http fetch GET 304 https://registry.npmjs.org/sockjs-client 30ms (from cache)
525 silly fetchPackageMetaData error for sockjs-client@0.0.0-unreleasable No matching version found for sockjs-client@0.0.0-unreleasable
526 warn shoe@0.0.15 had bundled packages that do not match the required version(s). They have been replaced with non-bundled versions.
527 verbose type version
528 verbose stack sockjs-client: No matching version found for sockjs-client@0.0.0-unreleasable
...
```

**Comment by ForkInSpace** [comment]
Having a similar issue after upgrading npm to 5.7.1
Any progress towards the solution ?

**Comment by mojoaxel** [comment]
This problem should be fixed in https://github.com/substack/shoe/issues/33

**Comment by mojoaxel** [comment]
Looks like this is an npm issue. A workaround would be to use an older version of npm:
```bash
npm install -g npm@5.6.0
```

**Comment by ForkInSpace** [comment]
yeah, reverted back to npm 5.6.0 and deleting node_modules and installing them from scratch helped. @mojoaxel thx anyways

**Comment by atufkas** [comment]
Same here:

```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for sockjs-client@0.0.0-unreleasable
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget
npm ERR! notarget It was specified as a dependency of 'shoe'
npm ERR! notarget

$ npm --version
5.7.1

$ node --version
v6.11.3
```

But I'm not haven't a dependency on tiny-care-terminal at all. It's obviously caused by `shoe`which is installed as a dependency of `phantom` which I depend on directly. 

Downgrading to npm@5.6.0 as considered by @mojoaxel helped me as a workaround!

**Comment by kerseyd27** [comment]
Having Xcode installed helped a coworker. I would install Xcode and open it to accept agreements. Then run rm -rf node_modules && npm install -g tiny-care-terminal

**Comment by mojoaxel** [comment]
It looks like this issue has been fixed in npm v6.0.0 :tada:
